src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -963,18 +1034,138 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
-                    ' ZZZ What's the representation in bound trees?
-                End If
-
-                Return ImmutableArray.Create(Of IMemberInitializer)()
+                Dim initializer = MemberInitializersMappings.GetValue(Me, Function(objCreationStmt)
+                                                                              Dim objInitializerExp As BoundObjectInitializerExpressionBase = Me.InitializerOpt
+                                                                              If objInitializerExp IsNot Nothing Then
+                                                                                  Dim builder = ArrayBuilder(Of IMemberInitializer).GetInstance(objInitializerExp.Initializers.Length)
+                                                                                  For Each memberAssignment In objInitializerExp.Initializers
+                                                                                      Dim assignment = TryCast(memberAssignment, BoundAssignmentOperator)
+                                                                                      Dim left = assignment?.Left
+                                                                                      If left IsNot Nothing Then
+                                                                                          Select Case left.Kind
+                                                                                              Case BoundKind.FieldAccess
+                                                                                                  builder.Add(New FieldInitializer(assignment.Syntax, DirectCast(left, BoundFieldAccess).FieldSymbol, assignment.Right))
+                                                                                              Case BoundKind.PropertyAccess
+                                                                                                  builder.Add(New PropertyInitializer(assignment.Syntax, DirectCast(left, BoundPropertyAccess).PropertySymbol.SetMethod, assignment.Right))
+                                                                                          End Select
+                                                                                      End If
+                                                                                  Next
+                                                                                  Return builder.ToArrayAndFree()
+                                                                              End If
+
+                                                                              Return New IMemberInitializer() {}
+                                                                          End Function)
+
+                Dim immBuilder = ArrayBuilder(Of IMemberInitializer).GetInstance(initializer.Length)
+                immBuilder.AddRange(initializer)
+                Return immBuilder.ToImmutableAndFree()
```

Comment:
Cache this array instead? FYI, there is an API that creates an ImmutableArray from a regular array, ImmutableArray.Create or something like that.